[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-api-keys

Avoid placing API keys in source code. Instead use build tools that insert the API key using environment variables similar to:

```js
const apiKey = process.env.GOOGLE_MAPS_API_KEY;
```

This pattern enables use of development keys and prevents use of production keys that may have higher quotas.

üìã This rule is enabled in `plugin:googlemaps/recommended`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
const apiKey = "AIza00000000000000000000000000000000000";
```

‚úîÔ∏è Examples of **correct** code:
```js
const apiKey = process.env.GOOGLE_MAPS_API_KEY;
const apiKey = "YOUR_API_KEY";
const apiKey = "AIzaSomeStringThatDoesntMatch";
```

## Resources

* [Rule source](/src/rules/no-api-keys.ts)
* [Test source](/src/rules/no-api-keys.test.ts)
