[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# place-fields

Use the `fields` option to limit the fields returned by the API and costs. Request to the Places API are billed by the fields that are returned. See [data-skus](https://developers.google.com/maps/documentation/places/web-service/usage-and-billing#data-skus) for more details.
> **Note**: This rule is not exhaustive and ignores `Autocomplete.setFields()`.

📋 This rule is enabled in `plugin:googlemaps/recommended`.

## Rule details

❌ Examples of **incorrect** code:
```js
const service = new google.maps.places.PlacesService();
const request = {place_id: 'foo'};
service.getDetails(request)

const service = new google.maps.places.PlacesService();
    service.getDetails({})

const service = new google.maps.places.PlacesService();
service.getDetails({...{bar: 'foo'}})

const service = new google.maps.places.Autocomplete(null, {});
const service = new google.maps.places.Autocomplete(null);
```

✔️ Examples of **correct** code:
```js
const service = new google.maps.places.PlacesService();
const request = {place_id: 'foo', fields: ['place_id']};
service.getDetails(request)

const service = new google.maps.places.PlacesService();
service.getDetails({place_id: 'foo', fields: ['place_id']})

const service = new google.maps.places.PlacesService();
service.getDetails({...{place_id: 'foo', fields: ['place_id']}})

const service = new google.maps.places.PlacesService();
service.getDetails({...{place_id: 'foo', 'fields': ['place_id']}})

const service = new google.maps.places.PlacesService();
const buildRequest = () => {};
service.getDetails(buildRequest())

const service = new google.maps.places.Autocomplete(null, {fields: ['place_id']});
```

## Resources

* [Rule source](/src/rules/place-fields.ts)
* [Test source](/src/rules/place-fields.test.ts)
