[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# require-js-api-loader

Require usage of [@googlemaps/js-api-loader](https://goo.gle/32rqTXJ) to dynamically [load the Google Maps API](https://goo.gle/3qipAT3).

üìã This rule is enabled in `plugin:googlemaps/recommended`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
const script = document.createElement('script');
script.src = 'https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap';

const url = 'https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap';
```

‚úîÔ∏è Examples of **correct** code:
```js
script.src = 'https://example.com'  // some other script hostname
```

## Resources

* [Rule source](/src/rules/require-js-api-loader.ts)
* [Test source](/src/rules/require-js-api-loader.test.ts)
